File: Data Structures & Algorithms/two-integer-sum/submission-2.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        # Two pointer solution
        vp = []
        for i in range(len(nums)):
            vp.append([nums[i], i])
        
        vp.sort()
        i, j = 0, len(nums) - 1
        while i < j:
            sum = vp[i][0] + vp[j][0]
            if sum == target:
                return [min(vp[i][1], vp[j][1]), max(vp[i][1], vp[j][1])]
            elif sum < target:
                i += 1
            else:
                j -= 1
        return []

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        if (nums.empty()) return 0;
        sort(nums.begin(), nums.end());
        int count = 0;
        int maxCount = 0;
        for (int i=1; i<nums.size(); i++) {
            if (nums[i] == nums[i-1] + 1) {
                count++;
            } else if (nums[i] == nums[i-1]) {
                continue;
            } else {
                count = 0;
            }
            if (count > maxCount) maxCount = count;
        }
        return maxCount + 1;
    }
};
